test/common/http/conn_pool_grid_test.cc

```diff
@@ -199,131 +208,154 @@ TEST_F(ConnectivityGridTest, FailureThenSuccessSerial) {
-                                       "reason", host_));
-  ASSERT_NE(grid_.second(), nullptr);
+      grid_->callbacks()->onPoolFailure(ConnectionPool::PoolFailureReason::LocalConnectionFailure,
+                                        "reason", host_));
+  ASSERT_NE(grid_->second(), nullptr);
 
   // onPoolReady should be passed from the pool back to the original caller.
-  ASSERT_NE(grid_.callbacks(), nullptr);
+  ASSERT_NE(grid_->callbacks(), nullptr);
   EXPECT_CALL(callbacks_.pool_ready_, ready());
   EXPECT_LOG_CONTAINS("trace", "second pool successfully connected to host 'hostname'",
-                      grid_.callbacks(1)->onPoolReady(encoder_, host_, info_, absl::nullopt));
-  EXPECT_TRUE(grid_.isHttp3Broken());
+                      grid_->callbacks(1)->onPoolReady(encoder_, host_, info_, absl::nullopt));
+  EXPECT_TRUE(grid_->isHttp3Broken());
 }
 
 // Test both connections happening in parallel and the second connecting.
 TEST_F(ConnectivityGridTest, TimeoutThenSuccessParallelSecondConnects) {
+  initialize();
   addHttp3AlternateProtocol();
-  EXPECT_EQ(grid_.first(), nullptr);
+  EXPECT_EQ(grid_->first(), nullptr);
 
   // This timer will be returned and armed as the grid creates the wrapper's failover timer.
   Event::MockTimer* failover_timer = new StrictMock<MockTimer>(&dispatcher_);
   EXPECT_CALL(*failover_timer, enableTimer(std::chrono::milliseconds(300), nullptr)).Times(2);
   EXPECT_CALL(*failover_timer, enabled()).WillRepeatedly(Return(false));
 
-  grid_.newStream(decoder_, callbacks_);
-  EXPECT_NE(grid_.first(), nullptr);
+  grid_->newStream(decoder_, callbacks_);
+  EXPECT_NE(grid_->first(), nullptr);
   EXPECT_TRUE(failover_timer->enabled_);
 
   // Kick off the second connection.
   failover_timer->invokeCallback();
-  EXPECT_NE(grid_.second(), nullptr);
+  EXPECT_NE(grid_->second(), nullptr);
 
   // onPoolFailure should not be passed up the first time. Instead the grid
   // should wait on the second pool.
   EXPECT_CALL(callbacks_.pool_failure_, ready()).Times(0);
-  grid_.callbacks()->onPoolFailure(ConnectionPool::PoolFailureReason::LocalConnectionFailure,
-                                   "reason", host_);
+  grid_->callbacks()->onPoolFailure(ConnectionPool::PoolFailureReason::LocalConnectionFailure,
+                                    "reason", host_);
 
   // onPoolReady should be passed from the pool back to the original caller.
-  EXPECT_NE(grid_.callbacks(), nullptr);
+  EXPECT_NE(grid_->callbacks(), nullptr);
   EXPECT_CALL(callbacks_.pool_ready_, ready());
-  grid_.callbacks(1)->onPoolReady(encoder_, host_, info_, absl::nullopt);
-  EXPECT_TRUE(grid_.isHttp3Broken());
+  grid_->callbacks(1)->onPoolReady(encoder_, host_, info_, absl::nullopt);
+  EXPECT_TRUE(grid_->isHttp3Broken());
+}
+
+// Test timer is affected by prior rtt.
+TEST_F(ConnectivityGridTest, SrttMatters) {
+  addHttp3AlternateProtocol(std::chrono::microseconds(2000000));
```

Comment:
nit: this really doesn't matter and you should ignore this but there are a lot of 0s there. How 'bout 2000 instead and then change for 4000 to 4 down below?